todoApp/spec/main.tsp

```diff
@@ -238,6 +238,20 @@ namespace TodoItems {
     @statusCode statusCode: 422;
   }
 
+  /**
+   * We need to decorate NotFoundResponse with an explicit `@error` decorator so it's clear that
+   * it is not a valid response
+   */
+  @error
```

Comment:
if this is an error, why there is no error body?

Reply:
That is the reason why I previously thought author may not intend this be an error.

There is already a generic `Standard4XXResponse` defined.
https://github.com/allenjzhang/typespec-e2e-demo/blob/25eb1d513debac3ecf54aaf7190aa6619945a25a/todoApp/spec/main.tsp#L140-L145
And a `alias WithStandardErrors<T> = T | Standard4XXResponse | Standard5XXResponse;`

If author want 404 be an error (with a code stating the reason), they don't need to write it here. The `WithStandardErrors` already covers all 4xx and 5xx.

But Brian has the final interpretation here. If he says it is error, it is error.

Reply:
Another thinking is if we should refine our http spec to conform to general customers' cases where 4xx should be treated as an error: https://github.com/microsoft/typespec/issues/5315

Reply:
There is no requirement that an error have a body, but I think this should probably have a specific `code` property like `not-found`, like the other explicitly defined errors.

Reply:
@bterlson do you think we should then make it with a body here like:
```
{
   "code": "not-found"
}
```

It would for sure simplify our life right now, as this is a scenario much more common

Reply:
Discussed with @bterlson , we should change this PR to:

```
@errormodel
model NotFoundErrorResponse {
   @statusCode statusCode: 404;
   code: "not-found";
}
```